src/coverlet.msbuild.tasks/BaseTask.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System;
+using Microsoft.Build.Utilities;
+
+namespace Coverlet.MSbuild.Tasks
+{
+    public abstract class BaseTask : Task
+    {
+        protected static IServiceProvider ServiceProvider { get; set; }
```

Comment:
@tonerdo take a look here, now to keep the semantics on statics we have a base class for tasks where we set "shared" static state for tasks, this is the only place where we need to have statics.